packages/process/src/node/process.ts

```diff
@@ -29,9 +29,11 @@ export abstract class Process {
     protected _killed = false;
 
     constructor(
-        @inject(ProcessManager) protected readonly processManager: ProcessManager,
+        protected readonly processManager: ProcessManager,
         protected readonly logger: ILogger,
-        readonly type: ProcessType) {
+        @unmanaged() readonly type: ProcessType,
```

Comment:
Why unmanaged ? is that needed since process won't be created direclty?

Reply:
The issue I hit is:

https://github.com/inversify/InversifyJS/issues/522

I tried making the Process class non-injectable (because it's never injected directly), but inversify complained.  Adding unmanaged seemed like the thing to do, according the the above mentionned issue.  However, I did not reall understand where we should apply unmanaged.  To all parameters?  Only to some?  What I did is I applied it to parameters that don't originally come from injected parameters (in subclasses).

Reply:
See: https://github.com/inversify/InversifyJS/blob/master/wiki/inheritance.md workaround C) and the intro before that.

So since inversify requires a derived class to have >= the number of constructor parameters than its base class.  The use of @unmanaged only works around that requirement.

So basically @unmanaged could by applied anywhere as long as it's applied to a sufficient number of parameters... its weird..

So in your case you could have it only on 2 of these...and it would work I think.